python/tvm/relay/frontend/pytorch.py

```diff
@@ -0,0 +1,974 @@
+
+        keepdims = int(inputs[2])
+        exclude = int(inputs[3])
+
+        return _op.mean(data, axis, keepdims, exclude)
+    return _impl
+
+def _chunk():
+    def _impl(inputs):
+        data = inputs[0]
+
+        num_chunks = int(inputs[1])
+        axis = int(inputs[2])
+
+        if isinstance(data, _expr.Var):
+            inferred_shape = get_tensor_from_relay_var(data)
+        elif isinstance(data, (_expr.Call, _expr.TupleGetItem)):
+            inferred_shape = _infer_type(data).checked_type.shape
+
+        shape = []
+        for infer in inferred_shape:
+            shape.append(infer)
+
+        dim = int(shape[axis])
+
+        if dim % num_chunks:
+            unif_size = int(dim / (num_chunks - 1))
+        else:
+            unif_size = int(dim / num_chunks)
+
+        chunks = []
+        for i in range(0, dim, unif_size):
+            begin = [0] * len(shape)
+            end = shape[:]
+            begin[axis] = i
+            end[axis] = i + unif_size
+            stride = [1] * len(shape)
+
+            chunk_out = _op.transform.strided_slice(data, begin, end, stride)
+            chunks.append(chunk_out)
+
+
+        if dim % num_chunks:
+            begin = [0] * len(shape)
+            end = shape[:]
+            begin[axis] = unif_size * (num_chunks - 1)
+            end[axis] = dim
+            stride = [1] * len(shape)
+
+            chunk_out = _op.transform.strided_slice(data, begin, end, stride)
+            chunks.append(chunk_out)
+
+        return chunks
+    return _impl
+
+def _matmul():
+    def _impl(inputs):
+        data0 = inputs[0]
+        data1 = inputs[1]
+        data1_t = _op.transpose(data1, axes=(1, 0))
+
+        return _op.nn.dense(data0, data1_t)
+    return _impl
+
+def _expand():
+    def _impl(inputs):
+        data_in = inputs[0]
+        if isinstance(data_in, _expr.Var):
+            shape = get_tensor_from_relay_var(data_in)
+        elif isinstance(data_in, (_expr.Call, _expr.TupleGetItem)):
+            shape = _infer_type(data_in).checked_type.shape
+
+        ndims = len(shape)
+        sizes = get_tensor_from_relay_var(inputs[1])
+        out = inputs[0]
+
+        for i in range(ndims):
+            if sizes[i] in {-1, shape[i]}:
+                continue
+            data = list()
+            for temp in range(sizes[i]):
+                data.append(out)
+            call = _op.tensor.concatenate(data, i)
+
+        return call
+    return _impl
+
+def _int():
+    def _impl(inputs):
+        if isinstance(inputs[0], _expr.Call):
+            return inputs[0]
+        return int(inputs[0])
+    return _impl
+
+def _listunpack():
+    def _impl(inputs):
+        return inputs[0]
+    return _impl
+
+def _to():
+    def _impl(inputs):
+        return inputs[0]
+    return _impl
+
+# Helper functions for operator implementation
+
+# Helper to grab actual tensor from a converted relay var (not sure if there's another way to do it)
+def get_tensor_from_relay_var(relay_var):
```

Comment:
Thanks, looks like infer_shape can be used everywhere I was using this helper as well as instead of manually grabbing the shape after calling infer_type.